src/runner/runESLint.js

```diff
@@ -96,8 +100,65 @@ const getComputedFixValue = ({ fix, quiet, fixDryRun }) => {
   return undefined;
 };
 
+const useFlatConfig = async () => {
+  if (semver.satisfies(version, '>=8')) {
+    switch (process.env.ESLINT_USE_FLAT_CONFIG) {
+      case 'true':
+        return true;
+      case 'false':
+        return false;
+      default:
+        return !!(await findUp('eslint.config.js', { cwd: process.cwd() }));
```

Comment:
IMO this is hard to read. Could you create a scope within the `default` case and extract `findUp` call to a variable